administrator/components/com_content/controllers/article.php

```diff
@@ -145,6 +146,39 @@ public function batch($model = null)
 	}
 
 	/**
+	 * Method to generate and store share token.
+	 *
+	 * @return  boolean   True if token successfully stored, false otherwise and internal error is set.
+	 *
+	 * @since   3.7
+	 */
+	public function shareDraft()
+	{
+		if (!JSession::checkToken('get'))
+		{
+			$app    = JFactory::getApplication();
+			$app->enqueueMessage(JText::_('JINVALID_TOKEN'));
+			echo new JResponseJson;
+			$app->close();
+		}
+		
+		try
+		{
+		$title = $this->input->getArray(array('jform' => array('title' => 'string')));
+
+		// Get the model
+		$model = $this->getModel();
+		$return = $model->shareToken($title);
+		echo new JResponseJson($return);
+		JFactory::getApplication()->close();
+		}
+		catch (Exception $e)
+		{
+		echo new JResponseJson($e);
+		}
+	}
```

Comment:
one doubt, $message= JText::_('MY_TASK_ERROR') and this would be called in echo new JResponseJson($reponse,$message,$error) and from where would the $error be fetched?